test/gtest/uct/ib/test_cqe_zipping.cc

```diff
@@ -0,0 +1,174 @@
+/**
+* Copyright (C) NVIDIA Corporation. 2022.  ALL RIGHTS RESERVED.
+*/
+
+#include <common/test.h>
+#include <common/test_helpers.h>
+#include <test_ib.h>
+
+extern "C" {
+#include <uct/api/uct.h>
+#include <uct/ib/base/ib_iface.h>
+}
+
+#include <vector>
+
+#if defined(ENABLE_STATS)
+#define UCT_INSTANTIATE_MLX5_TEST_CASE(_test_case) \
+    _UCT_INSTANTIATE_TEST_CASE(_test_case, rc_mlx5) \
+    _UCT_INSTANTIATE_TEST_CASE(_test_case, dc_mlx5) \
+    _UCT_INSTANTIATE_TEST_CASE(_test_case, ud_mlx5)
+
+struct test_cqe_zipping : public test_uct_ib_with_specific_port {
+    void send_while_possible()
+    {
+        /*
+         * The local deadline prevents hanging inside send loop in case of
+         * connectivity issues
+         */
+        const ucs_time_t deadline = ucs::get_deadline(15);
+        for (auto &peer : peers) {
+            while ((am_zcopy(peer) != UCS_ERR_NO_RESOURCE) &&
+                   (ucs_get_time() < deadline)) {
+                ++send_cnt;
+            }
+        }
+    }
+
+    void wait_for_completion()
+    {
+        /*
+         * The local deadline prevents hanging inside receive loop in case of
+         * connectivity issues
+         */
+        const ucs_time_t deadline = ucs::get_deadline(15);
+        while ((send_cnt != recv_cnt) && (ucs_get_time() < deadline)) {
+            progress();
+        }
+        ASSERT_EQ(send_cnt, recv_cnt);
+    }
+
+    size_t collect_zipped_counters()
```

Comment:
1. can it be const method?
2. rename to something like get_total_zipped_count()